helm/oauth2-proxy/templates/configmap-alpha.yaml

```diff
@@ -0,0 +1,39 @@
+{{- if .Values.alphaConfig.enabled }}
+{{- if not .Values.alphaConfig.existingConfig }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+{{- if .Values.alphaConfig.annotations }}
+  annotations:
+{{ toYaml .Values.alphaConfig.annotations | indent 4 }}
+{{- end }}
+  labels:
+    app: {{ template "oauth2-proxy.name" . }}
+{{- include "oauth2-proxy.labels" . | indent 4 }}
+  name: {{ template "oauth2-proxy.fullname" . }}-alpha
+data:
+  oauth2_proxy.yml: |
+    ---
+    server:
+      BindAddress: '0.0.0.0:4180'
+    {{- if .Values.metrics.enabled }}
+    metricsServer:
+      BindAddress: '0.0.0.0:44180'
+    {{- end }}
+    {{- if .Values.alphaConfig.providers }}
+    providers: {{ toYaml .Values.alphaConfig.providers | nindent 6 }}
+    {{- end }}
+    {{- if .Values.alphaConfig.upstreamConfig }}
+    upstreamConfig: {{ toYaml .Values.alphaConfig.upstreamConfig | nindent 6 }}
+    {{- end }}
+    {{- if .Values.alphaConfig.injectRequestHeaders }}
+    injectRequestHeaders: {{ toYaml .Values.alphaConfig.injectRequestHeaders | nindent 6 }}
+    {{- end }}
+    {{- if .Values.alphaConfig.injectResponseHeaders }}
+    injectResponseHeaders: {{- toYaml .Values.alphaConfig.injectResponseHeaders | nindent 6 }}
+    {{- end }}
+    {{- if .Values.alphaConfig.configData }}
+    {{- toYaml .Values.alphaConfig.configData | nindent 4 }}
```

Comment:
Should we just keep this one? What is the added values of the previous ones?

Reply:
the `configData` provides a future-proof solution, for 
changes to the oauth2-proxy config format which are not 
(yet) supported by the chart.

the other keys represent the currently supported 
configuration values and are provided to allow easier 
patching /chart-deployment. yes, in theory i could simply 
use `configData` as the only configuration input, but then
you would have to provide the entire config as a single 
blob (not really an issue, when you configure the chart 
deployment via files, but get cumbersome when you 
deploy it e.g. via CLI)

i can change the implementation, if you prefer to have the 
entire config under a single key

Reply:
i have reworked the PR to make no 
assumptions about the alpha config 
structure apart from the `server` and
`metricsServer` keys (which can be
enrichted using separate values)